src/main/java/cz/cvut/kbss/termit/persistence/dao/TermDao.java

```diff
@@ -699,6 +700,30 @@ public List<TermDto> findAllIncludingImported(String searchString, Vocabulary vo
         }
     }
 
+    @Override
+    public boolean exists(URI id) {
+        Objects.requireNonNull(id);
+        try {
+            URI vocabularyContext = null;
+            boolean notInAnyContext = false;
+            try {
+                vocabularyContext = contextMapper.getVocabularyContext(resolveTermVocabulary(id));
+            } catch (NoResultException e) {
+                notInAnyContext = true;
+            }
```

Comment:
I do not like catching the `NoResultException` here. Given the circumstances, I think it should be moved to `resolveTermContext` and when it is caught, a `NotFoundException` should be thrown, because it means the term does not exist in any vocabulary context.
Based on the current data storage architecture, it should never happen that a term is not in a vocabulary context.

Reply:
This change would also greatly simplify the rest of this method.

Reply:
I totally agree with that. I even had it done like that in the first place. But when running tests in `HtmlTermOccurrenceResolverTest` there are tests with data `rdfa-simple*.html` that don't follow that every term is in context. I tried to change the data to include the terms there in a context with `about `property. But was unsuccessful in even listing contexts present in the data with entityManager.

Reply:
Ok, I will update the tests first and let you know.